### ISSUE TITLE
\Yii::$app->formatter use current time (UTC) if time not set

### ISSUE THREAD

**Issue by bscheshirwork** [yiisoft/yii2#14202]
### What steps will reproduce the problem?
```php
        echo $dateStart . PHP_EOL;
        echo \Yii::$app->formatter->asDatetime($dateStart) . PHP_EOL;
        echo (new \DateTime($dateStart))->format('Y-m-d H:i:s') . PHP_EOL;
        echo `date`;
```
### What is the expected result?
```bash
2015-06-25
2015-06-25 00:00:00
2015-06-25 00:00:00
Wed May 24 17:36:20 MSK 2017
```
### What do you get instead?
```bash
2015-06-25
2015-06-25 14:36:20
2015-06-25 00:00:00
Wed May 24 17:36:20 MSK 2017
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | dev-master 44fa97f
| PHP version      | 7.1.5
| Operating system |alpine

**Comment by bscheshirwork** [comment]
```php
     /**
     * ...
     * - a string that can be [parsed to create a DateTime object](http://php.net/manual/en/datetime.formats.php).
     * ...
     */
```
https://github.com/yiisoft/yii2/blob/master/framework/i18n/Formatter.php#L680
!= "passed to create"

**Comment by bscheshirwork** [comment]
Must use |
http://php.net/manual/en/datetime.createfromformat.php

**Comment by cebe** [comment]
interesting, I thought we had fixed this already. Thanks for the detailed report.

**Comment by lynicidn** [comment]
hmmmm - what is started $dateStart?

**Comment by lynicidn** [comment]
why is bug ?

**Comment by lynicidn** [comment]
>Operating system 	alpine

???????

**Comment by bscheshirwork** [comment]
@lynicidn, пример в первом посте даёт ответ на все ваши вопросы. Также, возможно, будет полезным просмотреть ссылки.

**Comment by cebe** [comment]
fixed by #14203
